osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/openstack/securitygroup/SecurityGroupUpdateOrDeleteMetaTask.java

```diff
@@ -236,7 +240,19 @@ private void buildTaskGraph(EntityManager em, boolean isDeleteTg, String domainI
         VmDiscoveryCache vdc = new VmDiscoveryCache(this.sg.getVirtualizationConnector(),
                 this.sg.getVirtualizationConnector().getProviderAdminProjectName());
 
-        // in case of neutron sfc, make sure the SFC is upto date and SFC has the id set.
+		if (this.apiFactoryService.supportsNeutronSFC(this.sg.getVirtualizationConnector().getControllerType())) {
+
+			// if SFC binded and network element null -> create PC
+			// if SFC binded and network element not null -> update PC
+			// if SFC not binded and network element not null -> delete PC
+
+			if (this.sg.getServiceFunctionChain() != null
+					|| (this.sg.getServiceFunctionChain() == null && this.sg.getNetworkElementId() != null)) {
+				this.tg.appendTask(this.checkServiceFunctionChainMetaTask.create(this.sg),
```

Comment:
just check what is the predecessor and add the appropriate task guard.
ALL_ANCESTORS_SUCCEEDED taskguard is generally used only for delete objects tasks